scripts/backfill_errors.py

```diff
@@ -0,0 +1,427 @@
+        event.exception_stacks_type,
+        event.exception_stacks_value,
+        event.exception_stacks_mechanism_type,
+        event.exception_stacks_mechanism_handled,
+        event.exception_frames_abs_path,
+        event.exception_frames_colno,
+        event.exception_frames_filename,
+        event.exception_frames_function,
+        event.exception_frames_lineno,
+        event.exception_frames_in_app,
+        event.exception_frames_package,
+        event.exception_frames_module,
+        event.exception_frames_stack_level,
+        event.sdk_integrations,
+        event.modules_name,
+        event.modules_version,
+    )
+
+    return {
+        name: value
+        for name, value in zip(Error.field_names(), error._asdict().values())
+    }
+
+
+def backfill_errors() -> None:
+    events_storage = get_writable_storage(StorageKey.EVENTS)
+    errors_storage = get_writable_storage(StorageKey.ERRORS)
+
+    orderby_desc = "project_id DESC, timestamp DESC, event_id DESC"
+    orderby_asc = "project_id ASC, timestamp ASC, event_id ASC"
+
+    cluster = events_storage.get_cluster()
+
+    clickhouse = cluster.get_query_connection(ClickhouseClientSettings.MIGRATE)
+
+    events_table_name = events_storage.get_table_writer().get_schema().get_table_name()
+    errors_table_name = errors_storage.get_table_writer().get_schema().get_table_name()
+
+    try:
+        last_event_id, last_event_project, last_event_timestamp = clickhouse.execute(
+            f"""
+                SELECT replaceAll(toString(event_id), '-', ''), project_id, timestamp FROM {errors_table_name}
+                WHERE type != 'transaction'
+                AND NOT deleted
+                ORDER BY {orderby_asc}
+                LIMIT 1
+            """
+        )[0]
+        print(
+            f"Error data was found; starting migration from {last_event_id} {last_event_project} {last_event_timestamp}"
+        )
+    except IndexError:
+        last_event_id, last_event_project, last_event_timestamp = None, None, None
+
+    src_columns = ", ".join(
+        [cast(str, escape_identifier(field)) for field in Event.field_names()]
+    )
+
+    events_migrated = 0
+
+    while True:
+        event_conditions = ""
+        if last_event_id and last_event_project and last_event_timestamp:
+            event_conditions = f"""
+            AND (
+                (
+                    project_id = {escape_literal(last_event_project)}
+                    AND (
+                        timestamp < {escape_literal(last_event_timestamp)}
+                        OR (
+                            timestamp = {escape_literal(last_event_timestamp)}
+                            AND event_id < {escape_literal(last_event_id)}
+                        )
+                    )
+                )
+                OR (
+                    project_id < {escape_literal(last_event_project)}
+                )
+            )
+            """
+
+        raw_events = clickhouse.execute(
```

Comment:
If you are doing this while inserting data in the same table I think you should ensure you run the query with `in_order` load balancing to avoid querying from replicas that may not have received a block of data yet.

Reply:
This should be applying `load_balancing: in_order` already - similarly to all migrations (or places where we use the migrate Clickhouse settings) https://github.com/getsentry/snuba/blob/master/snuba/clusters/cluster.py#L35